unified_planning/io/pddl_grammar.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2021-2023 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from functools import partial
+from typing import Dict, List
+import typing
+import unified_planning as up
+import unified_planning.model.walkers
+from unified_planning.io.utils import set_parse_action, set_results_name, Located
+
+import pyparsing
+from pyparsing import ParseResults
+from pyparsing import CharsNotIn, Empty
+from pyparsing import Word, alphanums, alphas, ZeroOrMore, OneOrMore
+from pyparsing import Suppress, Group, Optional, Forward
+
+if pyparsing.__version__ < "3.0.0":
+    from pyparsing import oneOf as one_of
+    from pyparsing import restOfLine as rest_of_line
+else:
+    from pyparsing import one_of
+    from pyparsing import rest_of_line
+
+
+Object = "object"
+TypesMap = Dict[str, unified_planning.model.Type]
+
+
+def nested_expr():
```

Comment:
@Framba-Luca I am wondering, would it change anything if we avoid to call this function avery time (createing separate objects) and instead we "recycle" the same object in different points of the grammar?

Reply:
Actually, recycling the same object actually worsen performance by a lot. For now I didn't include it in the commit, but it can be investigated more deeply.